src/Core/Foundation/IoC/Container.php

```diff
@@ -104,9 +104,8 @@ private function makeInstanceFromClassName($className, array $alreadySeen)
 
         if ($classConstructor) {
             foreach ($classConstructor->getParameters() as $param) {
-                $paramClass = $param->getClass();
-                if ($paramClass) {
-                    $args[] = $this->doMake($param->getClass()->getName(), $alreadySeen);
+                if ($param->getType() && !$param->getType()->isBuiltin()) {
+                    $args[] = $this->doMake($param->getType()->getName(), $alreadySeen);
```

Comment:
As reported by PHPStan, `ReflectionType::getName()` does not exist
https://www.php.net/manual/en/class.reflectiontype.php

It seems you should use `__toString()` instead 👍

Reply:
Previously `$param->getType()` was checked if not null, now we check type 🚀